apis/v1alpha1/rhmi_conditions.go

```diff
@@ -0,0 +1,99 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"path"
+
+	addonv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	addoninstance "github.com/openshift/addon-operator/pkg/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RHMIConditionType string
+
+func (c RHMIConditionType) String() string {
+	return path.Join(group, string(c))
+}
+
+const HealthyConditionType RHMIConditionType = "Healthy"
+
+func (i *RHMI) InstalledCondition() metav1.Condition {
+	return addoninstance.NewAddonInstanceConditionInstalled(
+		metav1.ConditionTrue,
+		addonv1alpha1.AddonInstanceInstalledReasonSetupComplete,
+		"Installation complete",
+	)
+}
+
+func (i *RHMI) InstallBlockedCondition() metav1.Condition {
+	return addoninstance.NewAddonInstanceConditionInstalled(
+		metav1.ConditionFalse,
+		addonv1alpha1.AddonInstanceInstalledReasonBlocked,
+		"Installation blocked",
+	)
+}
+
+func (i *RHMI) UninstalledCondition() metav1.Condition {
+	return addoninstance.NewAddonInstanceConditionInstalled(
+		metav1.ConditionFalse,
+		addonv1alpha1.AddonInstanceInstalledReasonTeardownComplete,
+		"Teardown complete",
+	)
+}
+
+func (i *RHMI) UninstallBlockedCondition() metav1.Condition {
+	return addoninstance.NewAddonInstanceConditionInstalled(
+		metav1.ConditionTrue,
+		addonv1alpha1.AddonInstanceInstalledReasonBlocked,
+		"Teardown blocked",
+	)
+}
+
+func (i *RHMI) HealthyCondition() metav1.Condition {
+	return newRHMICondition(HealthyConditionType, metav1.ConditionTrue, "Healthy", "Core components healthy")
+}
+
+func (i *RHMI) UnHealthyCondition() metav1.Condition {
+	return newRHMICondition(HealthyConditionType, metav1.ConditionFalse, "Healthy", "One or more core components unhealthy")
+}
```

Comment:
Since we can't use the Healthy condition type provided by addon addon instance, I would like people's thought on whether we want our own `Healthy` conditions for core compoents since it's technically covered by the `Degraded` condition in terms of reporting. 

The only benefit currently to having this is to signify that core compoents is in a healthy state (phase complete). So we can have a status with `Healthy == True` and `Degraded == True` if for example, only Observability is down.

Reply:
I'm happy with this approach, think the condition descriptions are clear enough to distinguish the difference between healthy and degraded statuses.